beacon-chain/p2p/pubsub.go

```diff
@@ -132,7 +142,26 @@ func (s *Service) peerInspector(peerMap map[peer.ID]*pubsub.PeerScoreSnapshot) {
 //    Otherwise, set `message-id` to the first 20 bytes of the `SHA256` hash of
 //    the concatenation of `MESSAGE_DOMAIN_INVALID_SNAPPY` with the raw message data,
 //    i.e. `SHA256(MESSAGE_DOMAIN_INVALID_SNAPPY + message.data)[:20]`.
-func msgIDFunction(pmsg *pubsub_pb.Message) string {
+func (s *Service) msgIDFunction(pmsg *pubsub_pb.Message) string {
+	digest, err := ExtractGossipDigest(*pmsg.Topic)
```

Comment:
nil check of pmsg?

Reply:
this is applied inside pubsub, so it should not be possible to be nil, but we can be defensive over here.